dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/QueryControllerTest.java

```diff
@@ -0,0 +1,118 @@
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller;
+
+import static org.hisp.dhis.web.WebClientUtils.assertStatus;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.hisp.dhis.jsontree.JsonObject;
+import org.hisp.dhis.web.HttpStatus;
+import org.hisp.dhis.webapi.DhisControllerConvenienceTest;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests the {@link org.hisp.dhis.webapi.openapi.QueryController} with Mock MVC
+ * tests.
+ *
+ * @author Austin McGee
+ */
+class QueryControllerTest extends DhisControllerConvenienceTest
+{
+    private static final String API_QUERY_ALIAS = "/query/alias";
+
+    private static final String API_QUERY_ALIAS_REDIRECT = "/query/alias/redirect";
+
+    private static final String testTargetUrl = "/api/me?fields=id,username,surname,firstName";
+
+    private static final String testTargetAliasId = "671575eeea3d36a777efa6dcb48076083ff5cbbd";
+
+    private static final String testTargetAliasPath = API_QUERY_ALIAS + "/" + testTargetAliasId;
+
+    private static final String testTargetAliasFullPath = "/api" + testTargetAliasPath;
+
+    private static final String testTargetAliasHref = "http://localhost" + testTargetAliasFullPath;
+
+    @Test
+    void testGetUninitializedAlias()
+    {
+        assertStatus( HttpStatus.NOT_FOUND, GET( testTargetAliasPath ) );
+    }
+
+    @Test
+    void testCreateAlias()
+    {
+        JsonObject doc = POST( API_QUERY_ALIAS, "{ \"target\": \"" + testTargetUrl + "\" }" ).content();
+
+        assertTrue( doc.isObject() );
+
+        assertEquals( testTargetAliasId, doc.getString( "id" ).string() );
+        assertEquals( testTargetAliasFullPath, doc.getString( "path" ).string() );
+        assertEquals( testTargetAliasHref, doc.getString( "href" ).string() );
+        assertEquals( testTargetUrl, doc.getString( "target" ).string() );
+
+        JsonObject targetResponse = GET( testTargetUrl.substring( "/api".length() ) ).content();
+        JsonObject aliasResponse = GET( testTargetAliasPath ).content();
```

Comment:
This line fails with a 404 in the integration test.  It works well when executed manually.  It seems possible that there might be some odd / special handling of in-memory caches during test execution.